### PR TITLE
feat: [PL-32247]: tabbed interface for Authentication

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
### Summary

Implemented tabbed interface for Authentication page. Needed for new Authentication UI—with Tabs for each of {Authentication, User Management, IP Allowlist} without causing conflicts.
The new tabs will come up soon in upcoming feature PRs.

#### Screenshots

BEFORE:
image

AFTER:
image

Force-Merge reason: Coverage is failing as it's taking Authentication as new file, but it has all the old code.
The only new thing is Tabs.

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/33-auth-settings/pages/Configuration/Configuration.tsx [hunk 1]
```diff
@@ -6,120 +6,49 @@
...
+      <Page.Body className={css.configurationTabs}>
+        <Layout.Horizontal>
+          <Tabs
+            id="configurationTabs"
+            defaultSelectedTabId={defaultSelectedTabId}
+            onChange={newTabId => {
+              setView(newTabId as string)
+            }}
+          >
+            <Tab
```
Comment (sunnykesh-harness) [inline src/modules/33-auth-settings/pages/Configuration/Configuration.tsx:45]: Will we keep just one Tab  inside `Tabs` now?

File: src/modules/33-auth-settings/pages/Configuration/Configuration.tsx [hunk 2]
```diff
@@ -6,120 +6,49 @@
...
-              refetchAuthSettings={refetchAuthSettings}
-              canEdit={canEdit}
-              setUpdating={setUpdating}
+      <Page.Body className={css.configurationTabs}>
+        <Layout.Horizontal>
+          <Tabs
+            id="configurationTabs"
+            defaultSelectedTabId={defaultSelectedTabId}
+            onChange={newTabId => {
+              setView(newTabId as string)
```
Comment (sunnykesh-harness) [inline src/modules/33-auth-settings/pages/Configuration/Configuration.tsx:42]: Why we need this typecast?

**Review by sunnykesh-harness — APPROVED** [review]

File: src/modules/33-auth-settings/pages/Configuration/Authentication.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          (errorWhileFetchingAuthSettings?.data as Error)?.message ||
+          errorWhileFetchingAuthSettings?.message ||
+          (data?.resource ? undefined : getString('somethingWentWrong'))
+        }
+        retryOnError={() => refetchAuthSettings()}
+      >
+        {data?.resource && (
+          <React.Fragment>
+            {!canEdit && (
+              <Callout icon={null} className={css.callout}>
```
Comment (swarajcheguri-harness) [inline src/modules/33-auth-settings/pages/Configuration/Authentication.tsx:84]: why this ? will RBACTooltip does not take care ?

**Review by swarajcheguri-harness — APPROVED** [review]

File: src/modules/33-auth-settings/pages/Configuration/Configuration.tsx (hunk 1 above)
Comment (mayankVermaHarness) [reply src/modules/33-auth-settings/pages/Configuration/Configuration.tsx:45]: For now. Very soon, 2 new tabs are coming up.

File: src/modules/33-auth-settings/pages/Configuration/Authentication.tsx (hunk 3 above)
Comment (mayankVermaHarness) [reply src/modules/33-auth-settings/pages/Configuration/Authentication.tsx:84]: This PR made no change in the internals of the given Component except for moving it from `Configuration.tsx` to `Authentication.tsx`.

We can take this up separately.

File: src/modules/33-auth-settings/pages/Configuration/Configuration.tsx (hunk 2 above)
Comment (mayankVermaHarness) [reply src/modules/33-auth-settings/pages/Configuration/Configuration.tsx:42]: Fair point. Will need to make a change to use type `TabId` from Blueprint to remove this typecast. Can take this up in a separate PR.

**Pull request merged**